src/Data/Lens/Types.purs

```diff
@@ -27,21 +27,45 @@ import Data.Profunctor.Choice (class Choice)
 import Data.Profunctor.Closed (class Closed)
 import Data.Profunctor.Strong (class Strong)
 
--- | A general-purpose Data.Lens.
-type Optic p s t a b = p a b -> p s t
-type Optic' p s a = Optic p s s a a
+-- | Given a type whose "focus element" can always be retrieved,
+-- | a lens provides a convenient way to view, get, and transform
+-- | that element. 
```

Comment:
I'm not sure about the wording here, as it's not just that it can "always be retrieved", that's a consequence of the fact that a lens refers to a part of a structure that "always exists" - it focuses on part of a product type. Maybe the distinction is irrelevant though.

I assume "view, get" should be "view, set"? :smile: I'd probably say "modify" rather than "transform" too, but 🤷‍♂️

Reply:
Changed the "be retrieved" to "exists". I left "transform" because, to me, "modify" doesn't call to mind the idea of the type changing.